### ISSUE TITLE
threading does not obey limits

### ISSUE THREAD

**Issue by rbutleriii** [OpenGene/fastp#169]
I have been running with the default threads and explicitly setting `--thread 2` and in both cases, I see about 250-400% cpu usage. For parallelization that is an issue, as it appears the workers have access to more than one thread each.

**Comment by sfchen** [comment]
--thread 2 means use 2 processing worker threads.

Besides the worker threads, fastp uses one master thread, one loader thread, and one or two writer thread (depends on SE or PE)

**Comment by rbutleriii** [comment]
Fair enough. A feature request from me then would be to include that info in the documentation or in the `--thread` switch description. Conventional use of a thread switch usually defines the max threads it will consume for parallelization purposes.

**Comment by sfchen** [comment]
Thanks, I will mention this in the doc
